core/src/ledger/governance/utils.rs

```diff
@@ -196,12 +218,25 @@ impl TallyVote {
         }
     }
 
+    /// Check if a vote is nay
+    pub fn is_nay(&self) -> bool {
+        match self {
+            TallyVote::OnChain(vote) => vote.is_nay(),
+            TallyVote::Offline(vote) => vote.is_nay(),
+        }
+    }
+
     /// Check if two votes are equal
     pub fn is_same_side(&self, other: &TallyVote) -> bool {
-        let both_yay = self.is_yay() && other.is_yay();
-        let both_nay = !self.is_yay() && !other.is_yay();
-
-        both_yay || !both_nay
```

Comment:
@cwgoes here's the bug for the wrong e2e tally, it should have been `both_yay || both_nay`